org.adempiere.base/src/org/compiere/tools/FileUtil.java

```diff
@@ -486,11 +481,65 @@ public static File createTempFile(String prefix, String suffix, File directory)
 
         return f;	
 	}
+    
+    public static String getTempFolderName(File directory) throws IOException {
+    	Calendar cal = Calendar.getInstance();
+ 		SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmssSSS");
+ 		String dt = sdf.format(cal.getTime());
+ 		String tmpdirname = (directory != null) ? directory.getCanonicalPath() : System.getProperty("java.io.tmpdir");
+ 		tmpdirname += System.getProperty("file.separator") + "rpttmp_" + dt + "_" + Env.getContext(Env.getCtx(), Env.AD_SESSION_ID) + System.getProperty("file.separator");
+
+    	return tmpdirname;
+    }
 
     public static File createTempFile(String prefix, String suffix) throws IOException
     {
         return createTempFile(prefix, suffix, null);
     }
+    
+    /**
+     * Create a file with the given filename
+     * If the filename includes the path - creates the file as requested
+     * If it only includes the name, it creates the file in a safe-thread temporary folder  

```

Comment:
If you mouse over the method in Eclipse, you will notice the comments above doesn't looks right in the javadoc popup since line break is not preserve. To keep the line break, we have to add the <br/> tag. Also, it should be "thread safe" instead of "safe-thread".

Reply:
Changed